pylib/cqlshlib/copyutil.py

```diff
@@ -2101,6 +2101,13 @@ def convert_map(val, ct=cql_type):
             return ImmutableDict(frozenset((convert_mandatory(ct.subtypes[0], v[0]), convert(ct.subtypes[1], v[1]))
                                  for v in [split(split_format_str % vv, sep=sep) for vv in split(val)]))
 
+        def convert_vector(val, ct=cql_type):
+            string_coordinates = val.strip("[]").split(',')
+            if len(string_coordinates) != ct.vector_size:
+                raise ParseError("The length of given vector value '%d' is not equal to the vector size from the type definition '%d'" % (len(string_coordinates), ct.vector_size))
+            float_vector = [float(v) for v in string_coordinates]
```

Comment:
Converters for other types with subtypes (such as lists, sets, etc.) look to be using [convert_mandatory()](https://github.com/datastax/cassandra/blob/ds-trunk/pylib/cqlshlib/copyutil.py#L1940) for conversion of subtype data.  Probably a good idea to do the same here.

Reply:
Good point, the logic of conversion to `float` can be indeed handled by [convert_mandatory()](https://github.com/datastax/cassandra/blob/ds-trunk/pylib/cqlshlib/copyutil.py#L1940), also a nice side-effect of using this method is that it will handle other subtypes as well, that should address the remark you have made in https://github.com/datastax/cassandra/pull/866#discussion_r1396716570